### ISSUE TITLE
Not compatible to Thunderbird 91.0

### ISSUE THREAD

**Issue by Horst12** [kewisch/gdata-provider#194]
After upgrading to TB 91.0 all calendars are offiline and can't be enabled.
It was working fine with previous version TB 78.13

**Comment by chupsy80** [comment]
The calendar section works fine, after using the new calendar (+) button again and insert the used google e-Mail. The task-section is the one actually not working, unfortunately.

**Comment by thebertster** [comment]
Thunderbird 91.0 seems to support read/write access to Google Calendar natively (but also not working with Tasks). There are some mentions in the release notes of "Calendar: Remote calendar auto-detection now supported" but no specific mention of Google. It also works natively with Google contacts now!

If you add a Google account afresh, it will automatically discover Google calendars and contacts and import them.

As I upgraded, I had the same issue that my existing calendars were not working, but I removed those calendars and re-added as follows:

Add calendar, choose "On the network", enter Google e-mail address for "User name", enter "calendar.google.com" in "Location". I then get prompted to log in to Google and authorize Thunderbird to access my calendars. I'm then presented with the full list of my Google calendars (including calendars others have shared).

The calendar address once this is done is https://apidata.googleusercontent.com/caldav/v2/... rather than the googleapi:// URI that this extension used.

Only thing I did after that was go into each added calendar's properties and checked the appropriate e-mail address for invitation management (or None if not using invitations).

**Comment by racofer** [comment]
> Thunderbird 91.0 seems to support read/write access to Google Calendar natively (but also not working with Tasks) - I have now uninstalled this add-on and everything is working (so far, very well). There are some mentions in the release notes of "Calendar: Remote calendar auto-detection now supported" but no specific mention of Google. It also works natively with Google contacts now!

The main issue I'm facing with the native implementation of TB91 is that it doesn't allow me to add more than one calendar with the same name from different accounts, that is:
 - Gmail Account 1 - Contacts Calendar (added fine)
 - Gmail Account 2 - Contacts Calendar (grayed out, cannot add)
 - Gmail Account n - Contacts Calendar (grayed out, cannot add)

This also happens with any other calendar that shares a name with a previously added calendar from another account.

I never had this issue with the Provider for Google Calendar extension in TB 78 or 68.

**Comment by gumby-man** [comment]
> Add calendar, choose "On the network", enter Google e-mail address for "User name", enter "`calendar.google.com`" in "Location". I then get prompted to log in to Google and authorize Thunderbird to access my calendars. I'm then presented with the full list of my Google calendars (including calendars others have shared).

This does not work for me, „please wait“ message sticks forever. Any way to find problem? Or a way to get a working add-on? This really sucks.

**Comment by Dtrieb** [comment]
@gumby-man :

Add calendar, choose "On the network", enter Google e-mail address  without "**@gmail.com**" for User name, enter `https://apidata.googleusercontent.com/caldav/v2/` in "Location". 

That's it!

**Comment by djkraven** [comment]
I have a google calendar under other calendars in the Google listing that is not showing as an option to select. The calendar permissions are: Anyone can - See nothing, You can - See all event details. I tried @Dtrieb suggestion without success. It has a public URL of 
`https://calendar.google.com/calendar/embed?src=USER%40gmail.com&ctz=America%2FNew_York`
All the other calendars are showing as options to select using the TB 91 native and the user names only and location.  It was in TB 78 using the add-on. Any suggestions on how to get this calendar to show again?

**Comment by Dtrieb** [comment]
@djkraven so did it work for you now?

**Comment by djkraven** [comment]
@Dtrieb I removed the Provider addon and yes all my calendars now display. I just had to manually add the one that wasn't showing using the location example.

 The next thing to try is Google Reminders and Tasks. I have not used them in the past but would like to try. TB has a message to select a calendar that supports tasks. I have my calendar selected that I added a task to in the Google webpage defined but the message is still there.

**Comment by JPRuehmann** [comment]
Tried to uninstall The AddOn, delete the Callendars and reinstall them but it wont work.
After entering my Name and the Link above I get the Google Authentication (Username, Password, Code via Handy) but the I only get an Error That the Site can´t be connected. If I close that Window I get asked the Password and eveerything begins anew. There is no Change with or without AddOn.

How can I reenable mi Callendars?
Thanks.

**Comment by sba** [comment]
for me with `calendar.google.com` in location and the gmail-address as username it works perfect without the addon. with `https://apidata.googleusercontent.com/caldav/v2/` it does not work.

**Comment by JPRuehmann** [comment]
This Site can´t be found.

**Comment by JPRuehmann** [comment]
Will there be a working version for Thunderbird again?

**Comment by sba** [comment]
> 
> 
> Will there be a working version for Thunderbird again?

It works like a charm with the newly built-in google calendar-provide in Thunderbird 91. I think the add-on is no longer needed. i don't see any benefits with the add-on - Maybe you have to start with a new profile if there are problems with the calendar link...

**Comment by mrwensveen** [comment]
> > Will there be a working version for Thunderbird again?
> 
> It works like a charm with the newly built-in google calendar-provide in Thunderbird 91. I think the add-on is no longer needed. i don't see any benefits with the add-on - Maybe you have to start with a new profile if there are problems with the calendar link...

If this is true, it's not mentioned anywhere.

**Comment by Horst12** [comment]
> 
> 
> > Will there be a working version for Thunderbird again?
> 
> It works like a charm with the newly built-in google calendar-provide in Thunderbird 91. I think the add-on is no longer needed. i don't see any benefits with the add-on - Maybe you have to start with a new profile if there are problems with the calendar link...

You still need the Add-on for getting Tasks

**Comment by sba** [comment]
> 
> 
> > > Will there be a working version for Thunderbird again?
> > 
> > 
> > It works like a charm with the newly built-in google calendar-provide in Thunderbird 91. I think the add-on is no longer needed. i don't see any benefits with the add-on - Maybe you have to start with a new profile if there are problems with the calendar link...
> 
> You still need the Add-on for getting Tasks

Yes, this is true - tasks are not supported with built-in functionality. Never used, not noticed...

**Comment by racofer** [comment]
> It works like a charm with the newly built-in google calendar-provide in Thunderbird 91. I think the add-on is no longer needed. i don't see any benefits with the add-on - Maybe you have to start with a new profile if there are problems with the calendar link...

I'm unable to add some calendars from multiple Google accounts, like Holidays, Contacts (for birthdays), and others. If they share the same name on the remote server, even if they come from different Google accounts, I can add only one at a time. And even then, the imported calendars won't allow me to dismiss or snooze notifications. Clean installed TB 91 with the same results.

In the end I installed TbSync and now everything works.

**Comment by JPRuehmann** [comment]
No it is completely impossible to add Google callendars with Thunderbird since.

**Comment by Horst12** [comment]
> 
> 
> No it is completely impossible to add Google callendars with Thunderbirds Build in function
> After the Login it trys to display a Site from Localhost which it cannot access..

Then you selcted the wrong entry for adding a calendar.
It very easy to add yozr Google calendars uising your gmail address.

**Comment by Dtrieb** [comment]
🛑 Please stop writing now. 🛑

The plugin actually didn't work so Google Tasks, didn't work at the moment.
But if you only want use your calendar 🗓️ then it works without the plugin.

**Comment by JPRuehmann** [comment]
No it is not.
I get the =Auth Autentication,
Then it asks for the Callendar rights to permit and then it goes to a localhost page that is not working.

**Comment by CheariX** [comment]
I ran into the issue that the new thunderbird 91 does not show all shared calenders. 
I then stumbled upon this [link](https://www.google.com/calendar/syncselect) where you can select which calender show up in thunderbird.

**Comment by rsiuta** [comment]
Is there any way to run Google calendars on Thunderbird 78+ to work as before?
Once again ...
Currently, in Lighting, I can create events in all calendars, but it is not possible to invite participants to them. It is also not Google's fault, because in another e-mail program everything has been working without any problems for over a year.
I can see that not only I have such a problem and there is no solution.
Via CalDav, I can only subscribe to one (first calendar) in my account, but I would like all of them to be visible. Access to other calendars is done.

**Comment by kewisch** [comment]
Hey folks, please give this one a try: [gdata-provider-91.0.0pre1.zip](https://github.com/kewisch/gdata-provider/files/7071662/gdata-provider-91.0.0pre1.zip)

**Comment by JPRuehmann** [comment]
Authentication is not working (blank Screen)
![Screenshot_20210829_090126](https://user-images.githubusercontent.com/4321829/131241769-f08eb40a-5aeb-4398-b93c-1f792b53a909.png)

**Comment by JPRuehmann** [comment]
Disabled UBlock Origin, now it works.
Thanks.

**Comment by Horst12** [comment]
Authentication is not working (blank Screen)

**Comment by Reuzenman** [comment]
After installing "gdata-provider-91.0.0pre1" my calendar is working again.

**Comment by FabriceC01** [comment]
Hello,
I've downloaded the gdata-provider-91.0.0pre1.zip but how can I install it. it's not an xpi.
Can someone build it ?
thanks for all.

**Comment by mrwensveen** [comment]
> 
> 
> Hello,
> I've downloaded the gdata-provider-91.0.0pre1.zip but how can I install it. it's not an xpi.
> Can someone build it ?
> thanks for all.

You can rename the .zip file to a .xpi (zippy) file.

**Comment by Horst12** [comment]
> 
> 
> Hello,
> I've downloaded the gdata-provider-91.0.0pre1.zip but how can I install it. it's not an xpi.
> Can someone build it ?
> thanks for all.
Install Add-on from file accepts the zip file without any problem.
No need to do anything else.

**Comment by mattmiller747** [comment]
> Hey folks, please give this one a try: [gdata-provider-91.0.0pre1.zip](https://github.com/kewisch/gdata-provider/files/7071662/gdata-provider-91.0.0pre1.zip)

THANK YOU -  Worked like a charm !  PS  a M Bugzilla Moderator Gent said " That is not yet released and has a serious bug. Please remove it to. It has not even passed review!"

**Comment by field210** [comment]
will it work with google task in v91?

**Comment by jobisoft** [comment]
> PS a M Bugzilla Moderator Gent said " That is not yet released and has a serious bug. Please remove it to. It has not even passed review!"

That was me. The developer (kewisch) asked me to abort review of that version, as it has a bug. That is why it is still not available on addons.thunderbird.net.

**Comment by Reuzenman** [comment]
> 
> 
> will it work with google task in v91?

Yes, my calendar and task  working in 91.

**Comment by mattmiller747** [comment]
Hi again John

Well, I 've been testing it and it seems FINE

I did a post in TB and it Synced to my Phone

and I did the reverse and it Synced to TB


If you care to share with me the 'bug'

I'd be happy to see if it raises its ugly head with me !

On 8/31/2021 3:41 PM, John Bieling wrote:
>
>     PS a M Bugzilla Moderator Gent said " That is not yet released and
>     has a serious bug. Please remove it to. It has not even passed
>     review!"
>
> That was me. The developer (kewisch) asked me to abort review of that 
> version, as it has a bug. That is why it is still not available on 
> addons.thunderbird.net.
>
-- 
Adios,

Matt Miller


-- 
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus

**Comment by chilidad** [comment]
**Sync Google Tasks in Thunderbird 91...**

Very glad to see kewisch working to give TB ability to sync tasks with Google Calendar. This will finally get us past the "Please select a calendar that supports tasks" message showing the TB's Task Tab!

I've installed the abovementioned gdata-provider-91.0.0pre1.zip, and now see the "Google Calendar" option added to the Create New Calendar dialog box (Now shows "_On My Computer_" "_On the Network_" and "_Google Calendar_"). Unfortunately, I can't then get past the authentication login, which has a black screen as JPRuehmann and Horst12 mentioned above.

I'll definitely monitor this thread for further solutions. Thanks again!

**Comment by JPRuehmann** [comment]
And with this it won´t work anymore.

**Comment by jobisoft** [comment]
Could be that you cannot upgrade from the unofficial version. Remove calendars and add them anew.

**Comment by JPRuehmann** [comment]
I tyed to remove Callendars, remove the provider, reboot, install the new provider.
Won´t work.

**Comment by JPRuehmann** [comment]
Tried to remove the new and reinstall the old wont work to.
Empty window (only Add-On activated is the Google Provider.

**Comment by JPRuehmann** [comment]
![Screenshot_20210902_150526](https://user-images.githubusercontent.com/4321829/131850156-91bcd909-9cfb-4565-9fe3-5137c41942b5.png)

**Comment by JPRuehmann** [comment]
to which destination it should point?

**Comment by mattmiller747** [comment]
Hi John et al
        As a precaution, I removed the Prior 'bug' version, and then installed the new one and it worked just fine and dandy ! :-)
THANK YOU !!!

**Comment by JPRuehmann** [comment]
for me that won´t work , see above Screenshots.

**Comment by chilidad** [comment]
THANK YOU JOHN!!

_Note: I've successfully installed Provider_for_google_calendar-91.0.1, and have confirmed up/down syncing with my Google Calendar. FWIW, I've listed below my install log steps that worked for ME. Also listed a few minor bugs below._

Bugs
(See "MINORBUG1" below) In Calendars Tab, when adding, sometimes the "Google Calendar" option shows, sometimes it doesn't.
(See "MINORBUG2" below) In Task List properties, the refresh time should allow refreshing less than 30 minutes.
(See "MINORBUG3" below) Should be able to display tasks on Calendar grid. This is a preference rather than a bug, and it's probably TB core and not related to this Add-on.

**Syncing Google Tasks with TB**
![image](https://user-images.githubusercontent.com/43141154/131876998-bdc5d40d-79f5-441c-9372-79480927cf11.png)

**Comment by jobisoft** [comment]
I only did the review after @kewisch submitted. He is the one who did the work!

**Comment by chilidad** [comment]
Of course... my bad... ALSO THANKYOU PHILIPP KEWISCH!!!

**Comment by Horst12** [comment]
Thunderbird updated the Add-on from the pre version successfully to the 91.0.1
I had to allow access to the google data after the first restart.
No problems so far.

**Comment by JPRuehmann** [comment]
why it did not work for me? the last version before worked?

**Comment by rsiuta** [comment]
OK I have gdata-provider unzipped. How I can install this in TB?

**Comment by jobisoft** [comment]
Just search for google in the add-on manager to get the latest version of the povider. No need to manually install something.

**Comment by JPRuehmann** [comment]
go to the HamburgerMenue, AddOns, Klick on the Gear Symbol, select Install from File, select the .zip File and open.
If the zip File is not shown rename it to .xpi.

**Comment by rsiuta** [comment]
Ok, I have all callendars, but I'm still can't invite anybody do meeting for example.

**Comment by kewisch** [comment]
Hey Folks, I'm going to go ahead and close this issue since support for 91 is released. A few random comments I've caught:

@JPRuehmann I'm not sure why the localhost thing shows, the window should be closing by then and the Provider picking out the code. I think you filed a new issues on this, please do if you haven't. Error console messages appreciated!

@chilidad MINORBUG1 -> This doesn't sound great. Can you file a new issue with steps to reproduce, and error console messages for the case where it doesn't show? MINORBUG2 -> This is on purpose to manage the quota I have available for Google Calendar. Refreshing more often can cause some trouble. MINORBU3 -> There is an option for this (View -> Calendar -> Current View -> Tasks in Calendar)

If anyone else is experiencing issues please open a new one.
